.github/workflows/sqlite-build.yml

```diff
@@ -9,66 +9,144 @@ on:
 
 env:
   # Common SQLite compile-time options based on official documentation
+  # Source: https://www.sqlite.org/howtocompile.html
   SQLITE_COMPILE_TIME_OPTIONS: >-
     -O3
     -DSQLITE_ENABLE_FTS4
     -DSQLITE_ENABLE_FTS5
     -DSQLITE_ENABLE_RTREE
     -DSQLITE_ENABLE_GEOPOLY
-  
-  # Unix/Linux linking flags
-  UNIX_LDFLAGS: >-
-    -lpthread
-    -ldl
-    -lm
 
 jobs:
-  sqlite-build:
-    runs-on: ubuntu-latest
+  build:
+    strategy:
+      matrix:
```

Comment:
I still think matrix is not the best way to deal with this. Maybe separate jobs would be better.

Reply:
I'm all for separate workflows here as well.